drake/core/test/matrix_compare_test.cc

```diff
@@ -0,0 +1,152 @@
+#include "drake/util/eigen_matrix_compare.h"
+#include "gtest/gtest.h"
+
+using drake::util::MatrixCompareType;
+
+namespace drake {
+namespace test {
```

Comment:
Yes.  I still think an anonymous namespace is preferable to `namespace test`, but apparently there's no section of the style guide I can cite for that view, so I can live with your solution.